plugins/frontend/backstage-plugin-github-pull-requests/src/components/usePullRequests.ts

```diff
@@ -14,131 +14,130 @@
+import { useAsync } from 'react-use';
 import { githubPullRequestsApiRef } from '../api/GithubPullRequestsApi';
 import { useApi, githubAuthApiRef } from '@backstage/core-plugin-api';
-import { PullsListResponseData } from '@octokit/types';
+import { RequestError } from "@octokit/request-error";
 import moment from 'moment';
-import { PullRequestState } from '../types';
+import { PrState, PullRequest, PullRequestState } from '../types';
 import { useBaseUrl } from './useBaseUrl';
-
-export type PullRequest = {
-  id: number;
-  number: number;
-  url: string;
-  title: string;
-  updatedTime: string;
-  createdTime: string;
-  state: string;
-  draft: boolean;
-  merged: string|null;
-  creatorNickname: string;
-  creatorProfileLink: string;
-};
+import { useGithubPullRequests } from "./GithubPullRequestsContext"
 
 export function usePullRequests({
   owner,
   repo,
   branch,
   state,
+  pageSize,
 }: {
   owner: string;
   repo: string;
   branch?: string;
-  state?: PullRequestState;
+  state: PullRequestState;
+  pageSize?: number;
 }) {
   const api = useApi(githubPullRequestsApiRef);
   const auth = useApi(githubAuthApiRef);
   const baseUrl = useBaseUrl();
   const [total, setTotal] = useState(0);
   const [page, setPage] = useState(0);
-  const [pageSize, setPageSize] = useState(5);
+  const { prState, setPrState } = useGithubPullRequests()
   const getElapsedTime = (start: string) => {
     return moment(start).fromNow();
   };
 
-  const { loading, value: prData, retry, error } = useAsyncRetry<
-    PullRequest[]
-  >(async () => {
-    const token = await auth.getAccessToken(['repo']);
-    if (!repo) {
-      return [];
-    }
-    return (
-      api
-        // GitHub API pagination count starts from 1
-        .listPullRequests({
-          token,
-          owner,
-          repo,
-          pageSize,
-          page: page + 1,
-          branch,
-          state,
-          baseUrl,
-        })
-        .then(
-          ({
-            maxTotalItems,
-            pullRequestsData,
-          }: {
-            maxTotalItems?: number;
-            pullRequestsData: PullsListResponseData;
-          }) => {
-            if (maxTotalItems) {
-              setTotal(maxTotalItems);
-            }
+  const { loading, value, error } = useAsync(async (): Promise<PullRequest[] | void> => {
+    let result;
+    try {
+      const token = await auth.getAccessToken(['repo']);
+      const {
+        maxTotalItems,
+        pullRequestsData,
+        etag
+      } = await api.listPullRequests({
+        token,
+        owner,
+        repo,
+        pageSize,
+        page: page + 1,
+        branch,
+        state,
+        baseUrl,
+        etag: state && prState[state].etag || ""
+      })
+      if (maxTotalItems) {
+        setTotal(maxTotalItems);
+      }
+      if (etag) {
+        setPrState((current: PrState) => ({
+          ...current,
+          ...{ [state]: { ...current[state], etag } }
```

Comment:
Would this be the same written as:
```suggestion
          [state]: { ...current[state], etag }
```
?